extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/RequestScopeHandlerInterceptor.java

```diff
@@ -31,10 +32,14 @@ public RequestScopeHandlerInterceptor() {
         // This interceptor is called first, so, we should be on the event loop.
         Context capturedVertxContext = Vertx.currentContext();
         if (capturedVertxContext != null) {
+            InjectableContext.ContextState state = null;
```

Comment:
I think that you can make this var effectively final, i.e. `InjectableContext.ContextState state;` and assign `null` if `!activateAndDeactivateContext`.

Reply:
fixed